app/services/album.js

```diff
@@ -1,18 +1,34 @@
 const axios = require('axios'),
   config = require('../../config'),
   base_uri = config.common.albumsApi.url,
-  logger = require('../logger');
+  logger = require('../logger'),
+  helpers = require('../helpers');
 
-exports.getAlbum = params =>
+exports.allAlbums = params =>
   axios
-    .get(`${base_uri}/albums/${params.id}`)
+    .get(`${base_uri}/albums`)
+    .then(response => {
+      const { sortKey, sortOrder, filter } = params;
+      const dataAlbums = filter ? helpers.search(response.data, filter) : response.data;
+      const paginate = helpers.paginate(dataAlbums, params);
+      const albums = helpers.order(paginate, sortKey, sortOrder);
+      return albums;
+    })
+    .catch(error => {
```

Comment:
I think you should order and then paginate. If you want, for more readability, you can separate this task in the promise chain:
```javascript
...
.then(response => response.data)
.then(albums => filter ? helpers.search(albums, filter) : albums)
.then(albums => helpers.order(albums, sortKey, sortOrder))
.then(albums => helpers.paginate(albums, params);
```

Reply:
what a great solution